workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/ConnectionFlowImpls.cs

```diff
@@ -0,0 +1,269 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Improbable.Worker.CInterop;
+using Improbable.Worker.CInterop.Alpha;
+using Locator = Improbable.Worker.CInterop.Locator;
+using LocatorParameters = Improbable.Worker.CInterop.LocatorParameters;
+using AlphaLocator = Improbable.Worker.CInterop.Alpha.Locator;
+
+namespace Improbable.Gdk.Core
+{
+    public interface IConnectionFlowImpl
+    {
+        Task<Connection> CreateAsync(ConnectionParameters parameters);
+    }
+
+    public class ReceptionistFlowImpl : IConnectionFlowImpl
+    {
+        public string ReceptionistHost = "127.0.0.1";
+        public ushort ReceptionistPort = 7777;
+        public string WorkerId;
+
+        public ReceptionistFlowImpl(string workerId)
+        {
+            WorkerId = workerId;
+        }
+
+        public async Task<Connection> CreateAsync(ConnectionParameters parameters)
+        {
+            // TODO: Error handling.
+            using (var connectionFuture =
+                Connection.ConnectAsync(ReceptionistHost, ReceptionistPort, WorkerId, parameters))
+            {
+                return await Utils.TryToConnectAsync(connectionFuture).ConfigureAwait(false);
+            }
+        }
+    }
+
+    public class LocatorFlowImpl : IConnectionFlowImpl
+    {
+        public string LocatorHost = "locator.improbable.io";
+
+        public LocatorParameters LocatorParameters = new LocatorParameters
+        {
+            LoginToken = new LoginTokenCredentials
+            {
+                Token = string.Empty
```

Comment:
do we need to set all of those strings explicitly to `string.Empty`?

Reply:
We _may_ do. I didn't test what happens if you call `args.SetCommandLineValue(key, ref something.dest);` where `something.dest is null` but `something` is valid